src/access/AccessRestriction.cpp

```diff
@@ -0,0 +1,280 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AccessRestriction.h"
+
+#include <algorithm>
+#include <lib/core/Global.h>
+
+using namespace chip::Platform;
+
+namespace chip {
+namespace Access {
+
+std::vector<SharedPtr<AccessRestriction::Entry>> AccessRestriction::sCommissioningEntries;
+
+CHIP_ERROR AccessRestriction::CreateFabricEntries(const FabricIndex fabricIndex)
+{
+    for (auto & entry : sCommissioningEntries)
```

Comment:
why do all the fabrics match the commissioningARL?

This feels like example code that got mixed into the base implementation.

Reply:
When commissioning completes, the CommissioningARL entries are installed in the ARL attribute for the commissioning fabric.

Reply:
But this is not a requirement in the spec and it seems like the restrictions for commissioning would be more strict that the fabric restrictions. This is not an example implementation as far as I can tell - this is the handler for all devices. So having this here doesn't make sense.

Reply:
+1 please do not copy CommissioningARL to actual ARL

Reply:
Fixed/removed.  I misunderstood part of the spec.